examples/scripts/dpo_online.py

```diff
@@ -66,19 +80,20 @@
         padding_side="left",
         trust_remote_code=model_config.trust_remote_code,
     )
+    if tokenizer.chat_template is None:
+        tokenizer.chat_template = SIMPLE_QUERY_CHAT_TEMPLATE
+    
     dataset = load_dataset(args.dataset_name)
 
-    def process(row):
+    def prepare_dataset(row):
         row["prompt"] = tokenizer.apply_chat_template(row["prompt"], tokenize=False, add_generation_prompt=True)
         return row
 
-    # Compute that only on the main process for faster data processing.
-    # see: https://github.com/huggingface/trl/pull/1255
     with PartialState().local_main_process_first():
-        dataset = dataset.map(process, num_proc=training_args.dataset_num_proc)
+        dataset = dataset.map(prepare_dataset, num_proc=training_args.dataset_num_proc)
 
     prompts = dataset[args.dataset_test_split]["prompt"][:8]
-    log_completions_callback = LogCompletionsCallback(prompts)
+    log_completions_callback = LogCompletionsCallback(prompts, freq=training_args.logging_steps)
```

Comment:
I'd revert this change.
Having this makes it not compatible with logging_steps < 1.
When you set `freq=None`, the callback will get the "true" logging steps from the trainer state, see transformers.Trainer source code:

```python
        if args.logging_steps is not None:
            if args.logging_steps < 1:
                self.state.logging_steps = math.ceil(max_steps * args.logging_steps)
            else:
                self.state.logging_steps = args.logging_steps
```

Reply:
Ah nice catch! I'll revert

Reply:
Reverted in https://github.com/huggingface/trl/pull/1839/commits/8db4b718ff673408901485977b02b1849bc2619e